tests/testthat/test-DataUpgrade.R

```diff
@@ -0,0 +1,165 @@
+
+test_that("CreateTreatmentsDataFrame() parses perfect treatment string", {
+  treatments_string <- "Egg,Flour,Sugar,Butter,Cinnamon"
+  
+  treatment_df <- CreateTreatmentsDataFrame(treatments_string)
+  
+  expect_equal(colnames(treatment_df), c("Number", "Label"),
+               label = format_vector_to_string(colnames(treatment_df)))
+  expect_equal(nrow(treatment_df), 5,
+               label = nrow(treatment_df))
+  expect_equal(treatment_df$Number, 1:5,
+               label = format_vector_to_string(treatment_df$Number))
+  expect_equal(treatment_df$Label, c("Egg", "Flour", "Sugar", "Butter", "Cinnamon"),
+               label = format_vector_to_string(treatment_df$Label))
+})
+
+test_that("CreateTreatmentsDataFrame() parses string with internal, leading and trailing spaces", {
+  treatments_string <- " Egg , Flour , Sugar ,  Butter , Cinnamon  "
+  
+  treatment_df <- CreateTreatmentsDataFrame(treatments_string)
+  
+  expect_equal(colnames(treatment_df), c("Number", "Label"),
+               label = format_vector_to_string(colnames(treatment_df)))
+  expect_equal(nrow(treatment_df), 5,
+               label = nrow(treatment_df))
+  expect_equal(treatment_df$Number, 1:5,
+               label = format_vector_to_string(treatment_df$Number))
+  expect_equal(treatment_df$Label, c("Egg", "Flour", "Sugar", "Butter", "Cinnamon"),
+               label = format_vector_to_string(treatment_df$Label))
+})
+
+test_that("CreateTreatmentsDataFrame() parses string with duplicated and trailing commas", {
+  treatments_string <- "Egg,,Flour,Sugar,Butter,Cinnamon,"
+  
+  treatment_df <- CreateTreatmentsDataFrame(treatments_string)
+  
+  expect_equal(colnames(treatment_df), c("Number", "Label"),
+               label = format_vector_to_string(colnames(treatment_df)))
+  expect_equal(nrow(treatment_df), 5,
+               label = nrow(treatment_df))
+  expect_equal(treatment_df$Number, 1:5,
+               label = format_vector_to_string(treatment_df$Number))
+  expect_equal(treatment_df$Label, c("Egg", "Flour", "Sugar", "Butter", "Cinnamon"),
+               label = format_vector_to_string(treatment_df$Label))
+})
+
```

Comment:
Should there be another test for when there are spaces in the middle of an item, e.g. 'Baking Powder', as a treatment name like this would cause errors?

Reply:
I've added handling for this